### DISCUSSION TITLE
Queries Related Licenses Scanning

### DISCUSSION THREAD

**Discussion by parvjain639** [aquasecurity/trivy#4986] [Q&A] [answered]
### Question

We were trying to scan our local file YARN.LOCK as it gives proper vulnerability report but unable to list licenses. 
so can you explain how to list licenses for local file??

Also, we found this in documentation, after using (--license-full) still not getting output for licenses. 

Currently, the standard license scanning doesn't support filesystem and repository scanning.
License scanning	Image	Rootfs	Filesystem	Repository
Standard                 	✅	           ✅	       -	                -
Full (--license-full)	✅	           ✅	     ✅	              ✅

But, still not resolving the issue.
Following is the command and the output for the file we have scanned.

COMMAND: trivy fs <FOLDER NAME> --scanners license --license-full
OUTPUT:      INFO    Full license scanning is enabled

As we have tried same with Docker Images also, we got proper outputs for Licenses!!
Can you help us to get the Complete license details of local file scanning??

### Target

Filesystem

### Scanner

License

### Output Format

None

### Mode

Standalone

### Operating System

Debian 11 (Bullseye)

### Version

```bash
Version: 0.44.1
Vulnerability DB:
                 Version: 2
```

**Comment by nikpivkin** [comment, marked as answer]
Hi @parvjain639 !

At the moment, Trivy does not support license scanning in projects with `yarn`.

But there is an [issue](https://github.com/aquasecurity/trivy/issues/4811) for adding this feature. This feature will be included in the next trivy release.

**Reply by parvjain639** [reply]
Thank you so much for your quick response!!

Can you please share a list of supported File formats, if available??

We have tried scanning package.json, package-lock.json, go.mod, go.sum, etc. but not getting license details.

**Reply by nikpivkin** [reply]
@parvjain639 You can find out for which languages license scanning is supported on their respective pages. https://aquasecurity.github.io/trivy/v0.44/docs/scanner/vulnerability/language/

**Reply by nikpivkin** [reply]
License scanning is supported for package-lock and go.mod. Could you give an example when it doesn't work for you?

**Reply by parvjain639** [reply]
Can you please share your email-id so that we can share the testing file??

**Reply by nikpivkin** [reply]
@parvjain639 Could you join [Slack](https://slack.aquasec.com/) and write to me in DM (`@Nikita Pivkin`)?

**Comment by parvjain639** [comment]
> https://aquasecurity.github.io/trivy/v0.44/docs/scanner/vulnerability/language/
Is this same for license scanning too??
